components/logger.rst

```diff
@@ -159,6 +159,28 @@ Configuration options:
 -  **tag** (*Optional*, string): The tag (seen in front of the message in the logs) to print the message
    with. Defaults to ``main``.
 
+Logger Automation
+-----------------
+
+.. _logger-on_message:
+
+``on_message``
+**************
+
+This automation will be triggered when a new message is added to the log.
+In :ref:`lambdas <config-lambda>` you can get the message, log level and tag from the trigger
+using ``message``, ``level`` and ``tag``.
```

Comment:
For this trigger it would be useful to add the type information here - most triggers use the C++ std::string class for ease of use. Logger is an exception to this (because dynamic memory std::string uses is not ideal).

Maybe add the C type here in parentheses (`const char *`, `int`)